packages/core/content-manager/server/src/services/uid.ts

```diff
@@ -53,36 +53,51 @@ export default ({ strapi }: { strapi: Strapi }) => ({
     value: string;
     locale: string;
   }) {
-    const foundDocuments = await strapi.documents(contentTypeUID).findMany({
-      filters: {
-        [field]: value,
-      },
-      locale: locale ?? null,
-      // TODO: Check UX. When modifying an entry, it only makes sense to check for collisions with other drafts
-      // However, when publishing this "available" UID might collide with another published entry
-      status: 'draft',
-    });
+    const modelKind = strapi.getModel(contentTypeUID)?.kind;
+
+    let foundDocuments;
+
+    if (modelKind === 'singleType') {
+      const document = await strapi.documents<Schema.SingleType>(contentTypeUID).find({
+        filters: {
+          [field]: value,
+        },
+        locale: locale ?? null,
+        status: 'draft',
+      });
+      foundDocuments = document ? [document] : [];
+    } else {
+      foundDocuments = await strapi.documents<Schema.CollectionType>(contentTypeUID).findMany({
+        filters: {
+          [field]: value,
+        },
+        locale: locale ?? null,
+        // TODO: Check UX. When modifying an entry, it only makes sense to check for collisions with other drafts
+        // However, when publishing this "available" UID might collide with another published entry
+        status: 'draft',
+      });
+    }
```

Comment:
These are the type of cases that could add a bit more friction cc @alexandrebodin

Reply:
Nice usecase to consider indeed 👍